test/unit/specs/store/search-store.spec.js

```diff
@@ -240,238 +297,144 @@ describe('Search Store', () => {
         shouldPersistMedia: false,
         mediaType: IMAGE,
       }
-      const action = store.actions(audioServiceMock, imageServiceMock)[
-        FETCH_MEDIA
-      ]
-      action({ commit, dispatch, state }, params)
+      const action = store.actions(services)[FETCH_MEDIA]
+      await action({ commit, dispatch, state }, params)
 
-      expect(dispatch).not.toHaveBeenLastCalledWith('SEND_SEARCH_QUERY_EVENT', {
+      expect(dispatch).not.toHaveBeenCalledWith('SEND_SEARCH_QUERY_EVENT', {
         query: params.q,
         sessionId: state.usageSessionId,
       })
     })
 
-    it('FETCH_COLLECTION_IMAGES on success', (done) => {
-      const params = {
-        provider: 'met',
-        page: 1,
-        shouldPersistMedia: false,
-        mediaType: IMAGE,
+    it('FETCH_MEDIA on error', async () => {
+      const mediaType = IMAGE
+      services[IMAGE] = {
+        search: jest.fn(() => Promise.reject('error')),
       }
-      const action = store.actions(audioServiceMock, imageServiceMock)[
-        FETCH_COLLECTION_IMAGES
-      ]
-      action({ commit, dispatch }, params).then(() => {
-        expect(commit).toBeCalledWith(FETCH_START_MEDIA, { mediaType: IMAGE })
-        expect(commit).toBeCalledWith(FETCH_END_MEDIA, { mediaType: IMAGE })
-
-        expect(commit).toBeCalledWith(SET_MEDIA, {
-          media: searchData.results,
-          mediaCount: searchData.result_count,
-          shouldPersistMedia: params.shouldPersistMedia,
-          page: params.page,
-          mediaType: IMAGE,
-        })
-
-        const newParams = { ...params, source: params.provider }
-        delete newParams.provider
-        expect(imageServiceMock.getProviderCollection).toBeCalledWith(newParams)
-        done()
-      })
-    })
-
-    it('FETCH_COLLECTION_IMAGES calls search API if q param exist', (done) => {
       const params = {
-        q: 'nature',
-        provider: 'met',
+        q: 'foo',
         page: 1,
         shouldPersistMedia: false,
-        mediaType: IMAGE,
+        mediaType,
       }
-      const action = store.actions(audioServiceMock, imageServiceMock)[
-        FETCH_COLLECTION_IMAGES
-      ]
-      action({ commit, dispatch }, params).then(() => {
-        const newParams = { ...params, source: params.provider }
-        delete newParams.provider
-        expect(imageServiceMock.search).toBeCalledWith(newParams)
-
-        done()
-      })
-    })
-
-    it('FETCH_COLLECTION_IMAGES calls getProviderCollection API if li param exist', (done) => {
-      const params = {
-        li: 'by',
-        provider: 'met',
-        page: 1,
-        shouldPersistMedia: false,
-        mediaType: IMAGE,
+      const action = store.actions(services)[FETCH_MEDIA]
+      try {
+        await action({ commit, dispatch, state }, params)
+      } catch (error) {
+        expect(commit).toHaveBeenCalledWith(FETCH_START_MEDIA, { mediaType })
+        expect(dispatch).toHaveBeenCalledWith('HANDLE_IMAGE_ERROR', error)
```

Comment:
Same thing here as above, we should use `expect().rejects` (which can be awaited so the subsequent assertions can be made)